src/web/api.rs

```diff
@@ -279,6 +285,112 @@ impl ApiService {
+                    StatusCode::BAD_REQUEST,
+                );
+            }
+
+            let serve_dom = serve_tree.inner();
+
+            let packed_uuid = Uuid::new_v4().to_simple().to_string();
+            let model_path = format!("{}.rbxm", packed_uuid);
+
+            let studio = match RobloxStudio::locate() {
+                Ok(studio) => studio,
+                Err(error) => {
+                    return json(
+                        ErrorResponse::bad_request(format!(
+                            "Couldn't find Roblox Studio: {}",
+                            error
+                        )),
+                        StatusCode::INTERNAL_SERVER_ERROR,
+                    );
+                }
+            };
+
+            // TODO: memofs does not have create_dir_all, but it is necessary here.
+            let packed_path = studio
+                .content_path()
+                .join(CREATE_ASSETS_DIR)
+                .join(&model_path);
+
+            let mut writer: Vec<u8> = Vec::new();
+
+            let export_tree = WeakDom::new(
+                InstanceBuilder::new("Folder")
+                    .with_name("Assets")
+                    .with_children(request.assets.iter().copied().filter_map(|asset_ref| {
+                        let exporting = match serve_dom.get_by_ref(asset_ref) {
+                            Some(instance) => instance,
+                            None => {
+                                log::warn!(
+                                    "Received asset ID for an instance not in our tree ({})",
+                                    asset_ref
+                                );
+
+                                return None;
+                            }
+                        };
+
+                        // TODO: Children of instance are not used here, as Rojo might already have
+                        // created them, and can place them inside instead.
+                        Some(
+                            InstanceBuilder::new("Folder")
+                                .with_name(asset_ref.to_string())
+                                .with_child(
+                                    InstanceBuilder::new(&exporting.class)
+                                        .with_name(&exporting.name)
+                                        .with_properties(exporting.properties.to_owned()),
+                                ),
+                        )
+                    })),
+            );
+
+            if let Err(error) = rbx_binary::to_writer_default(
+                &mut writer,
+                &export_tree,
+                &vec![export_tree.root_ref()],
+            ) {
+                return json(
+                    ErrorResponse::bad_request(format!("Couldn't write DOM to buffer: {}", error)),
+                    StatusCode::INTERNAL_SERVER_ERROR,
+                );
+            }
+
+            if let Err(error) = vfs.write(packed_path, writer) {
```

Comment:
This shouldn't go through the VFS; we should use regular filesystem methods here.